cpp/src/io/json/json_column.cu

```diff
@@ -531,6 +532,63 @@ void make_device_json_column(device_span<SymbolT const> input,
       }
     });
 
+  // 4. scatter List offset
+  // copy_if only node's whose parent is list, (node_id, parent_col_id)
+  // stable_sort by parent_col_id of {node_id}.
+  // For all unique parent_node_id of (i==0, i-1!=i), write start offset.
+  //                                  (i==last, i+1!=i), write end offset.
+  //    unique_copy_by_key {parent_node_id} {row_offset} to
+  //    col[parent_col_id].child_offsets[row_offset[parent_node_id]]
+
+  auto& parent_col_ids = sorted_col_ids;  // reuse sorted_col_ids
```

Comment:
I'd prefer a copy here, any use of `sorted_col_ids` after alterations to `parent_col_ids` could lead to a bug. I'm more worried about such a bug than the cost of memcpy, but folks might disagree.
Maybe a future refactor (where each step is a function) could facilitate this reuse.

Reply:
This is done to reduce memory usage.
Right now, for 1 GB JSON string, it takes 4.93 GB during`reduce_to_column_tree`. It's less compared to peak memory 8.9 GB.

Reply:
I'm not quite sold on it, it's only an integer per column :)